### PR TITLE
feat(rust): complete secure channel implementation, fix router bugs

### PR DESCRIPTION
**Pull request by robinbudd**

### CODE REVIEW

File: implementations/rust/build.gradle [hunk 1]
```diff
@@ -42,7 +42,8 @@ task cargoFmt {
 task cargoClippy {
   doLast {
     exec {
-      commandLine 'cargo', 'clippy', '--all', '--', '-D', 'warnings'
+      //commandLine 'cargo', 'clippy', '--all', '--', '-D', 'warnings'
```
Comment (mikelodder7) [inline implementations/rust/build.gradle:45]: Did we want to keep this commented out?

File: implementations/rust/c/generate_bindings/build.rs [hunk 2]
```diff
@@ -81,13 +82,12 @@ fn main() {
 
     // Generate bindings if llvm-config is present
     if let Some(llvm_config) = llvm_config {
+        println!("cargo:warning=Test1");
```
Comment (mikelodder7) [inline implementations/rust/c/generate_bindings/build.rs:85]: probably don't need this line anymore

File: implementations/rust/c/generate_bindings/build.rs [hunk 3]
```diff
@@ -81,13 +82,12 @@ fn main() {
...
         let src_file = root.join("implementations/rust/c/bindings/src/bindings.rs");
         generate_bindings(llvm_config, include_dirs, &src_file);
     } else {
-        println!(
-            "cargo:error={}",
-            "LLVM_PREFIX was not set, and cannot find llvm-config, will not regenerate bindings"
-        );
+        println!("cargo:warning=Test2");
```
Comment (mikelodder7) [inline implementations/rust/c/generate_bindings/build.rs:89]: Or this one.

**Review by mikelodder7 — CHANGES_REQUESTED** [review]
Looks good. Just some suggestions. Just rebase then I can approve.

File: implementations/rust/build.gradle (hunk 1 above)
Comment (robinbudd) [reply implementations/rust/build.gradle:45]: not any more :)

**Review by mrinalwadhwa — APPROVED** [review]

**Pull request merged**